### PR TITLE
Adjust quality java plugin for subprojects, database plugin

### PR DESCRIPTION
**Pull request by shalauniou**

### CODE REVIEW

File: src/main/groovy/com/epam/petclinic/plugin/QualityAwareJavaPlugin.groovy [hunk 1]
```diff
@@ -35,44 +35,45 @@ public class QualityAwareJavaPlugin implements Plugin<Project> {
 
     @Override
     void apply(Project project) {
-        project.plugins.apply(JAVA_PLUGIN_ID)
-
         project.extensions.create(QualityAwareJavaExtension.NAME, QualityAwareJavaExtension)
+        project.childProjects.values().each { subProject ->
```
Comment (Andrei-Khadziukou) [inline src/main/groovy/com/epam/petclinic/plugin/QualityAwareJavaPlugin.groovy]: why don't use `subprojects` for quality plugin? Something like that

subprojects {
apply plugin: 'java'
...
}

File: src/main/groovy/com/epam/petclinic/plugin/QualityAwareJavaPlugin.groovy [hunk 2]
```diff
@@ -35,44 +35,45 @@ public class QualityAwareJavaPlugin implements Plugin<Project> {
...
-            configureCheckStyle(project)
-            configurePMD(project)
-            configureFindBugs(project)
+            subProject.afterEvaluate {
+                configureCheckStyle(subProject)
+                configurePMD(subProject)
+                configureFindBugs(subProject)
+            }
         }
+        //TODO: enable fail build after fixing tons of code quality errors
```
Comment (Andrei-Khadziukou) [inline src/main/groovy/com/epam/petclinic/plugin/QualityAwareJavaPlugin.groovy:65]: why don't disable them via properties files in each project?

File: src/main/groovy/com/epam/petclinic/plugin/QualityAwareJavaPlugin.groovy (hunk 1 above)
Comment (shalauniou) [reply src/main/groovy/com/epam/petclinic/plugin/QualityAwareJavaPlugin.groovy]: We have task `checkCodeQualityErrors` for all subprojects it gathers errors. I think that is for such purposes if we have to combine all subprojects and run the task from the root.

Comment (Andrei-Khadziukou) [reply src/main/groovy/com/epam/petclinic/plugin/QualityAwareJavaPlugin.groovy]: Could you please analyze both approaches to discuss them?

**Pull request merged**